primitives/tests/api.rs

```diff
@@ -0,0 +1,283 @@
+// SPDX-License-Identifier: CC0-1.0
+
+//! Test the API surface of `primitives`.
+//!
+//! The point of these tests are to check the API surface as opposed to test the API functionality.
+//!
+//! ref: <https://rust-lang.github.io/api-guidelines/about.html>
+
+#![allow(dead_code)]
+#![allow(unused_imports)]
+#![cfg(feature = "alloc")]
+#![cfg(feature = "arbitrary")]
+
+use arbitrary::Arbitrary;
+use bitcoin_primitives::block::{Checked, Unchecked};
+use bitcoin_primitives::script::{self, ScriptHash, WScriptHash};
+use bitcoin_primitives::{
+    absolute, block, merkle_tree, pow, relative, transaction, OutPoint, Script, ScriptBuf,
+    Sequence, Transaction, TxIn, TxOut, Txid, Witness, Wtxid,
+};
+use hashes::sha256t;
+
+/// A struct that includes all public non-error enums.
+#[derive(Debug)] // All public types implement Debug (C-DEBUG).
+struct Enums {
+    a: absolute::LockTime,
+    b: relative::LockTime,
+    // block::Checked, // Empty enums are not constructable.
+    // block::Unchecked,
```

Comment:
In d845272ddcf2ebe907a63eefaaeb504b59210f52:

We should uncomment these. They don't need to be constructible for the API tests.

Reply:
Now resolved.